pkg/bgpv1/manager/reconcilerv2/service.go

```diff
@@ -302,6 +320,70 @@ func (r *ServiceReconciler) getDesiredSvcRoutePolicies(p ReconcileParams, desire
 	return desiredSvcRoutePolicies, nil
 }
 
+// handleOverlappingDesiredSvcRoutePolicies handles the case of a desired BGP advertisement that overlaps a previous
+// selector match. Overlaps are handled as additive operations for attributes that can be unioned (standard and large
+// communities). If local preference was set, it remains set to the first advertisement matched.
+//
+// This function modifies existingPolicy to contain the unions described above.
+func (r *ServiceReconciler) handleOverlappingDesiredSvcRoutePolicies(
+	existingPolicy *types.RoutePolicy,
+	candidatePolicy *types.RoutePolicy,
+	advert v2alpha1.BGPAdvertisement,
+) {
+
+	// For tests, we must maintain the original order.  Communities will hold
+	// the unique set of standard communities defined across both policies.
+	communities := []string{}
+	communitiesSeen := sets.New[string]()
+
+	largeCommunities := []string{}
+	largeCommunitiesSeen := sets.New[string]()
+
+	// Extract the existing policy's communities
+	for _, statement := range existingPolicy.Statements {
+		communities = append(communities, statement.Actions.AddCommunities...)
+		communitiesSeen.Insert(statement.Actions.AddCommunities...)
+
+		largeCommunities = append(largeCommunities, statement.Actions.AddLargeCommunities...)
+		largeCommunitiesSeen.Insert(statement.Actions.AddLargeCommunities...)
+	}
+	logMessagePreviousCommunities := fmt.Sprint(communities)
+	logMessagePreviousLargeCommunities := fmt.Sprint(largeCommunities)
+
+	// Extract this policy's communities
+	for _, statement := range candidatePolicy.Statements {
```

Comment:
@harsimran-pabla As you look at this, I'm curious to hear if other ideas come to your mind that would be more efficient?  I feel like I'm iterating way too many times in this function.

As a reminder, I observed this code path being executed for each Service (ClusterIP, Loadbalancer, etc.) running on a given Node.  If a node has 100s of services, I wouldn't want this to code path to slow down reconciliation.

Because of that, I changed the log message below from `r.Logger.Warnf()` to `Debugf()`.